laurent22/joplin
# Mobile: Fixes #11264: Fix editor shows nothing when there are no selected note IDs

# Summary

This pull request fixes #11264 by preventing `reducer.ts` from clearing `selectedNoteIds` on mobile. Previously, some actions could result in the note the user was editing being removed from the selection.  This caused the note editor to vanish.

To preserve the existing behavior on desktop, this pull request  adds a property to the app state, `allowSelectionInOtherFolders`. When set to `true`, the reducer should avoid clearing `selectedNoteIds` when the selection is moved out of the current folder.

# Notes

- A previous attempt at a fix was merged into `release-3.1`. Due to an incorrectly resolved merge conflict, this fix was not applied to `dev`. This original fix worked by overriding the `noteId` prop passed to the note editor with the last non-null `noteId`. With this change, however, parts of the app that rely on `state.selectedNoteIds` to determine the note open in the editor remained broken when `selectedNoteIds` was cleared.
- When the note viewer is closed, `selectedNoteIds` may be set to the ID of the last-opened note. 

# Testing

On web:
1. Create notes A, B, C in different folders.
2. Create links from C to B to A.
3. Open the folder containing note C.
4. Open note C, follow the link to note B, follow the link to note A.
5. On a different Joplin client, move note B to a different folder.
6. Sync the other Joplin client.
7. Sync the original client (without leaving the note).
8. Press the back button.
9. Add a tag to note B.
   - The tag dialog should work.
10. On the other Joplin client, move note B back to folder B.
11. Sync both clients (without leaving note B).
12. On the original client, verify that note B is still open.
   - **Note**: The folder picker may still display the original folder.
13. Still on the original client, add a new tag to note B.
   - Verify that the tag dialog still works.
14. Open the notes list, select multiple notes, and move them to a different folder.
15. Verify that the notes have been moved.

On Android 13:
1. Follow a link to a note in a different folder.
2. Navigate to the device's app library.
3. Switch back to Joplin.
4. Verify that the same note is still open.
5. Move the note to a different folder.
6. Repeat steps 2-5.
